common/base/include/claraparabricks/genomeworks/utils/allocator.hpp

```diff
@@ -251,7 +255,7 @@ class CachingDeviceAllocator
         return static_cast<pointer>(ptr);
     }
 
-    /// \brief Asynchronously dealllocates allocated array
+    /// \brief Asynchronously dealllocates allocated array, calls cudaStreamSynchronize() on associated streams
```

Comment:
How about changing this to a very vague "may call cudaStreamSynchronize()"?

Reply:
yes, I missed this one

Reply:
done